projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -6509,21 +6523,29 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
     public endEdit(commit = true, event?: Event) {
         const row = this.crudService.row;
         const cell = this.crudService.cell;
-
+        let canceled = false;
         // TODO: Merge the crudService with with BaseAPI service
         if (!row && !cell) { return; }
 
         if (row?.isAddRow) {
             this.endAdd(commit, event);
             return;
         }
-        commit ? this.gridAPI.submit_value() : this.gridAPI.escape_editMode();
 
-        if (!this.rowEditable || this.rowEditingOverlay && this.rowEditingOverlay.collapsed || !row) {
-            return;
+        if (commit) {
+            canceled = this.gridAPI.submit_value();
+            if (canceled) {
+                return true;
+            }
+        } else {
+            this.crudService.exitCellEdit();
         }
 
-        this.endRowTransaction(commit, row);
+        canceled = this.crudService.exitRowEdit(commit);
+        if (canceled) {
+            this.crudService.rowEditingBlocked = true;
```

Comment:
This doesn't get cleared anywhere - if you cancel row edit once, after exiting edit mode you can't enter again